File: client/python/consumer.py
from kafka import KafkaConsumer

consumer = KafkaConsumer(
    'belajarkafka',
    bootstrap_servers='localhost:9092',
    group_id='nodejs',
    auto_offset_reset='earliest',
    enable_auto_commit=True
)

print("Listening for messages...")
for message in consumer:
    key = message.key.decode() if message.key else None
    value = message.value.decode() if message.value else None
    print(f"topic: {message.topic}, partition: {message.partition}")
    print(f"key: {key}, value: {value}")

File: client/consumer.js
import { Kafka } from "kafkajs";

const kafka = new Kafka({
    brokers: ["localhost:9092"]
});

const consumer = kafka.consumer({ 
    groupId: "nodejs" 
});

await consumer.subscribe({ 
    topic: "belajarkafka", 
    fromBeginning: true 
});

await consumer.connect();

await consumer.run({
    eachMessage: async ({ topic, partition, message }) => {
        const key = message.key ? message.key.toString() : null;
        const value = message.value ? message.value.toString() : null;
        console.log(`topic: ${topic}, partition: ${partition}`);
        console.log(`key: ${key}, value: ${value}`);
    }
});